app/org/sagebionetworks/bridge/upload/UploadArtifactsHandler.java

```diff
@@ -46,6 +46,7 @@ public void handle(@Nonnull UploadValidationContext context) {
         HealthDataRecordBuilder recordBuilder = context.getHealthDataRecordBuilder();
         HealthDataRecord record = recordBuilder.build();
         String recordId = healthDataService.createOrUpdateRecord(record);
+        context.setRecordId(recordId);
```

Comment:
This should be a recordGuid. In particular, you can migrate this record between data stores without an issue, it's not an auto-generated integer (or what we'd probably call an id).